ddtrace/profiling/profiler.py

```diff
@@ -276,9 +276,9 @@ def copy(self):
             service=self.service, env=self.env, version=self.version, tracer=self.tracer, tags=self.tags
         )
 
-    def start(self):
+    def _start(self):
+        # type: () -> None
         """Start the profiler."""
-        super(_ProfilerInstance, self).start()
```

Comment:
Seems like the profiler `start` was already thread-safe because of this call?

Reply:
No, the lock is released by the time `super` returns 😢 
In practice I don't think anyone ever tried to start a `Profiler` in two different threads, but hey that's still a bug if we want a `Service` to be thread-safe 😉

Reply:
Oh right of course the race was actually in `profiler.start` and not the super! 👍